lib/src/main/java/io/ably/lib/transport/SecureSSLSocketFactory.java

```diff
@@ -0,0 +1,89 @@
+package io.ably.lib.transport;
+
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This is a decorator for the {@code SSLSocketFactory} which modifies the enabled TLS protocols
+ * for each created {@code SSLSocket} to only use the protocols which are considered to be secure.
+ * <p>
+ * This class was created because the {@code SSLContext.getInstance()} method does not allow specifying
+ * precisely which TLS protocols can be used and which cannot.
+ */
+public class SecureSSLSocketFactory extends SSLSocketFactory {
```

Comment:
I wonder if there might be a better name for this class, given expansion of SSL would mean it's called "Secure Secure Sockets Layer Socket Factory". 😆 

Maybe `StricterSSLSocketFactory`. 🤔

Reply:
Good point :joy: What do you think about `SafeSSLSocketFactory` or `ProtectedSSLSocketFactory`? :thinking: I would like to leave some word that would hint that this factory is more secure than the regular one, if possible :wink:

Reply:
https://github.com/ably/ably-java/pull/754/commits/ba62780721d7f6e23bd25a65bc21900acc3a8eab